velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp

```diff
@@ -0,0 +1,313 @@
+    return false;
+  }
+
+  void initializeNewGroups(
+      char** groups,
+      folly::Range<const vector_size_t*> indices) override {
+    setAllNulls(groups, indices);
+    for (auto i : indices) {
+      new (groups[i] + offset_) BloomFilterAccumulator(allocator_);
+    }
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    decodeArguments(rows, args);
+    rows.applyToSelected([&](vector_size_t row) {
+      VELOX_USER_CHECK(
+          !decodedRaw_.isNullAt(row),
+          "First argument of bloom_filter_agg cannot be null");
+      auto group = groups[row];
+      auto tracker = trackRowSize(group);
+      auto accumulator = value<BloomFilterAccumulator>(group);
+      accumulator->init(capacity_);
+      accumulator->insert(decodedRaw_.valueAt<int64_t>(row));
+    });
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    VELOX_CHECK_EQ(args.size(), 1);
+    decodedIntermediate_.decode(*args[0], rows);
+    rows.applyToSelected([&](auto row) {
+      if (UNLIKELY(decodedIntermediate_.isNullAt(row))) {
+        return;
+      }
+      auto group = groups[row];
+      auto tracker = trackRowSize(group);
+      auto serialized = decodedIntermediate_.valueAt<StringView>(row);
+      auto accumulator = value<BloomFilterAccumulator>(group);
+      accumulator->mergeWith(serialized);
+    });
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    decodeArguments(rows, args);
+    auto tracker = trackRowSize(group);
+    auto accumulator = value<BloomFilterAccumulator>(group);
+    accumulator->init(capacity_);
+    if (decodedRaw_.isConstantMapping()) {
+      VELOX_USER_CHECK(
+          !decodedRaw_.isNullAt(0),
+          "First argument of bloom_filter_agg cannot be null");
+      // All values are same, just do for the first.
+      accumulator->insert(decodedRaw_.valueAt<int64_t>(0));
+      return;
+    }
+    rows.applyToSelected([&](vector_size_t row) {
+      VELOX_USER_CHECK(
+          !decodedRaw_.isNullAt(row),
+          "First argument of bloom_filter_agg cannot be null");
+      accumulator->insert(decodedRaw_.valueAt<int64_t>(row));
+    });
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    VELOX_CHECK_EQ(args.size(), 1);
+    decodedIntermediate_.decode(*args[0], rows);
+    auto tracker = trackRowSize(group);
+    auto accumulator = value<BloomFilterAccumulator>(group);
+    rows.applyToSelected([&](auto row) {
+      if (UNLIKELY(decodedIntermediate_.isNullAt(row))) {
```

Comment:
This implementation is not consistent with addIntermediateResults. Shouldn't we throw here?

Reply:
It consistents with addIntermediateResults, I run the gluten unit test, it can run into here